File: src/main/java/cspfj/heuristic/Lexico.java
package cspfj.heuristic;

import cspfj.problem.Variable;

public final class Lexico implements ValueHeuristic {

	public Lexico() {
	}

	public double getScore(final Variable var, final int index) {
		return index;
	}

	public String toString() {
		return "lexico";
	}

	@Override
	public void compute() {
		// Nothing to compute
	}

	@Override
	public int selectIndex(final Variable variable) {
		return variable.getFirst();
	}
}

File: src/main/scala/cspfj/heuristic/Lexico.scala
package cspfj.heuristic;

import cspfj.Variable;

final class Lexico extends ValueHeuristic {

  def score(variable: Variable, index: Int) = -index

  override def toString = "lexico";

  def compute() {
    // Nothing to compute
  }

  override def selectIndex(variable: Variable) = variable.dom.last

}
